helper/helper.go

```diff
@@ -183,11 +183,28 @@ func createPipe(pipe string) error {
 		return err
 	}
 
-	pipesMap[pipe] = true
+	pipesMap[pipe] = false
 	return nil
 }
 
 func deletePipe(pipe string) error {
+	if utils.FileExists(pipe) && !pipesMap[pipe] {
+		var err error
+		var handle *os.File
+		if *backupAgent {
+			handle, err = os.OpenFile(pipe, os.O_RDONLY|unix.O_NONBLOCK, os.ModeNamedPipe)
+		} else {
+			handle, err = os.OpenFile(pipe, os.O_WRONLY|unix.O_NONBLOCK, os.ModeNamedPipe)
+		}
```

Comment:
It would be more readable if make a flag depending on *backupAgent and use a single call to os.OpenFile, i.e.:
```
var int flag
if *backupAgent {
   flag = os.O_RDONLY|unix.O_NONBLOCK
} else {
  flag = os.O_WRONLY|unix.O_NONBLOCK
}
handle, err = os.OpenFile(pipe, flag, os.ModeNamedPipe)
```

Reply:
applied